src/python/dxpy/scripts/dx_build_app.py

```diff
@@ -1022,15 +1022,21 @@ def get_project_to_check():
         if "project" in extra_args:
             return extra_args["project"]
         if destination:
-            dest_project_id, _, _ = parse_destination(destination)
+            # "." is the default value set for args.destination which is
+            # resolved to the workspace container when run from the worker.
+            # For the purpose of feature switch check, we will use the project
+            # context ID (which requires VIEW project access for the job
+            # that builds the pipeline)
+            if destination == "." or destination.startswith("container-"):
+                return dxpy.PROJECT_CONTEXT_ID
+            else:
+                dest_project_id, _, _ = parse_destination(destination)
             return dest_project_id
         else:
             return dxpy.PROJECT_CONTEXT_ID
 
     dest_project_to_check = get_project_to_check()
-    features = dxpy.DXHTTPRequest("/" + dest_project_to_check + "/checkFeatureAccess", {"features": ["dxNextflow"]}).get("features", {})
-    # Expecting output {'features': {'dxNextflow': True}}
-    
+    features = dxpy.DXHTTPRequest("/" + dest_project_to_check + "/checkFeatureAccess", {"features": ["dxNextflow"]}).get("features", {})    
```

Comment:
> .. envvar:: DX_JOB_ID
   Should only be present if run in an Execution Environment; indicates
   the ID of the currently running job.

```suggestion
    if "DX_JOB_ID" in environ:
        dest_project_to_check = dxpy.PROJECT_CONTEXT_ID
        
    features = dxpy.DXHTTPRequest("/" + dest_project_to_check + "/checkFeatureAccess", {"features": ["dxNextflow"]}).get("features", {})    
```

Reply:
or
    ```
    if dxpy.JOB_ID:
        dest_project_to_check = dxpy.PROJECT_CONTEXT_ID
    ````

Reply:
What if the destination is a different project? I.e. if somebody runs this from jupyterlab

I think being a little bit defensive and parsing the destination does not hurt.